### ISSUE TITLE
Building blueprint not working

### ISSUE THREAD

**Issue by andreash** [palantir/blueprint#4652]
I can't get the build to work. I am not sure what I am doing wrong. Node & Python version down in the log.

#### Environment

- __Package version(s)__: 
- __Operating System__: Ubunutu 20.04
- __Browser name and version__: n/a

#### Steps to reproduce

1. git clone https://github.com/palantir/blueprint
1. yarn

#### Actual behavior
andreas@novi:~/SW-Projekte/test$ git clone https://github.com/palantir/blueprint
Klone nach 'blueprint' ...
remote: Enumerating objects: 42596, done.
remote: Counting objects: 100% (1086/1086), done.
remote: Compressing objects: 100% (661/661), done.
remote: Total 42596 (delta 559), reused 703 (delta 413), pack-reused 41510
Empfange Objekte: 100% (42596/42596), 98.33 MiB | 10.80 MiB/s, fertig.
Löse Unterschiede auf: 100% (32762/32762), fertig.
andreas@novi:~/SW-Projekte/test$ yarn
Debugger attached.
yarn install v1.22.10
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
Done in 0.08s.
Waiting for the debugger to disconnect...
andreas@novi:~/SW-Projekte/test$ cd blueprint
andreas@novi:~/SW-Projekte/test/blueprint$ yarn
Debugger attached.
yarn install v1.22.10
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
warning better-handlebars@0.1.1: The engine "atom" appears to be invalid.
warning language-less@0.34.3: The engine "atom" appears to be invalid.
info fsevents@2.1.3: The platform "linux" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning "lerna > @lerna/version > @lerna/github-client > @octokit/rest > @octokit/plugin-request-log@1.0.2" has unmet peer dependency "@octokit/core@>=3".
warning " > stylelint-config-palantir@5.0.0" has unmet peer dependency "stylelint@^13.3.2".
warning "stylelint-config-palantir > stylelint-config-standard@20.0.0" has unmet peer dependency "stylelint@>=10.1.0".
warning "stylelint-config-palantir > stylelint-order@4.1.0" has unmet peer dependency "stylelint@^10.0.1 || ^11.0.0 || ^12.0.0 || ^13.0.0".
warning " > stylelint-scss@3.18.0" has unmet peer dependency "stylelint@^8.0.0 || ^9.0.0 || ^10.0.0 || ^11.0.0 || ^12.0.0 || ^13.0.0".
warning "stylelint-config-palantir > stylelint-config-standard > stylelint-config-recommended@3.0.0" has unmet peer dependency "stylelint@>=10.1.0".
warning " > @blueprintjs/core@3.43.0" has unmet peer dependency "react@^15.3.0 || 16 || 17".
warning " > @blueprintjs/core@3.43.0" has unmet peer dependency "react-dom@^15.3.0 || 16 || 17".
warning " > @blueprintjs/datetime@3.22.1" has unmet peer dependency "react@^15.3.0 || 16 || 17".
warning " > @blueprintjs/datetime@3.22.1" has unmet peer dependency "react-dom@^15.3.0 || 16 || 17".
warning " > @blueprintjs/docs-theme@3.9.10" has unmet peer dependency "react@^15.3.0 || 16 || 17".
warning " > @blueprintjs/docs-theme@3.9.10" has unmet peer dependency "react-dom@^15.3.0 || 16 || 17".
warning " > @blueprintjs/popover2@0.5.0" has unmet peer dependency "react@^16.8.0 || ^17".
warning " > @blueprintjs/select@3.15.10" has unmet peer dependency "react@^15.3.0 || 16 || 17".
warning " > @blueprintjs/select@3.15.10" has unmet peer dependency "react-dom@^15.3.0 || 16 || 17".
warning " > @blueprintjs/table@3.8.28" has unmet peer dependency "react@^15.3.0 || 16 || 17".
warning " > @blueprintjs/table@3.8.28" has unmet peer dependency "react-dom@^15.3.0 || 16 || 17".
warning " > @blueprintjs/test-commons@0.10.11" has unmet peer dependency "react@>=15.3.0".
warning " > @blueprintjs/timezone@3.8.10" has unmet peer dependency "react@^15.3.0 || 16 || 17".
warning " > @blueprintjs/timezone@3.8.10" has unmet peer dependency "react-dom@^15.3.0 || 16 || 17".
warning " > @blueprintjs/tslint-config@3.0.4" has unmet peer dependency "tslint@^6.0.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/core > sass-inline-svg@1.2.3" has unmet peer dependency "node-sass@^4.5.3".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/core > webpack-cli@3.3.12" has unmet peer dependency "webpack@4.x.x".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/docs-app > copy-webpack-plugin@6.4.0" has unmet peer dependency "webpack@^4.37.0 || ^5.0.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/docs-app > webpack-dev-server@3.11.0" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/eslint-config > @typescript-eslint/eslint-plugin-tslint@4.9.1" has unmet peer dependency "tslint@^5.0.0 || ^6.0.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/karma-build-scripts > istanbul-instrumenter-loader@3.0.1" has unmet peer dependency "webpack@^2.0.0 || ^3.0.0 || ^4.0.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/karma-build-scripts > karma-webpack@5.0.0" has unmet peer dependency "webpack@^5.0.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/test-commons > enzyme-adapter-react-15@1.4.2" has incorrect peer dependency "react@^15.5.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/test-commons > enzyme-adapter-react-15@1.4.2" has unmet peer dependency "react-dom@^15.5.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/test-commons > enzyme-adapter-react-15@1.4.2" has unmet peer dependency "react-test-renderer@^15.5.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/test-commons > enzyme-adapter-react-16@1.15.5" has unmet peer dependency "react-dom@^16.0.0-0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/webpack-build-scripts > fork-ts-checker-notifier-webpack-plugin@3.0.0" has incorrect peer dependency "fork-ts-checker-webpack-plugin@^5.0.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/docs-app > webpack-dev-server > webpack-dev-middleware@3.7.2" has unmet peer dependency "webpack@^4.0.0".
warning "workspace-aggregator-80171510-5901-4ea2-86e5-cabd203a7cd6 > @blueprintjs/docs-data > @documentalist/compiler > typedoc@0.19.2" has incorrect peer dependency "typescript@3.9.x || 4.0.x".
[5/5] Building fresh packages...
[-/9] ⠈ waiting...
[8/9] ⠈ oniguruma
[6/9] ⠈ deasync
[4/9] ⠈ node-sass
error /home/andreas/SW-Projekte/test/blueprint/node_modules/oniguruma: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/andreas/SW-Projekte/test/blueprint/node_modules/oniguruma
Output:
Debugger attached.
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@12.22.1 | linux | x64
gyp info find Python using Python version 2.7.18 found at "/usr/bin/python2.7"
gyp info spawn /usr/bin/python2.7
gyp info spawn args [
gyp info spawn args   '/home/andreas/SW-Projekte/test/blueprint/packages/webpack-build-scripts/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/andreas/SW-Projekte/test/blueprint/node_modules/oniguruma/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/andreas/SW-Projekte/test/blueprint/packages/webpack-build-scripts/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/andreas/.cache/node-gyp/12.22.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/andreas/.cache/node-gyp/12.22.1',
gyp info spawn args   '-Dnode_gyp_dir=/home/andreas/SW-Projekte/test/blueprint/packages/webpack-build-scripts/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/andreas/.cache/node-gyp/12.22.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/andreas/SW-Projekte/test/blueprint/node_modules/oniguruma',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
Debugger attached.
Waiting for the debugger to disconnect...
gyp: Call to 'node -e "require('nan')"' returned exit status 0 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/home/andreas/SW-Projekte/test/blueprint/packages/webpack-build-scripts/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Linux 5.4.0-72-generic
gyp ERR! command "/home/andreas/.nvm/versions/node/v12.22.1/bin/node" "/home/andreas/SW-Projekte/test/blueprint/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /home/andreas/SW-Projekte/test/blueprint/node_modules/oniguruma
gyp ERR! node -v v12.22.1

#### Expected behavior

yarn should install all needed dependecies withouth error.

**Comment by andreAbtahi96** [comment]
Would this have anything to do with it?

"The platform "linux" is incompatible with this module."

**Comment by andreash** [comment]
> Would this have anything to do with it?
> 
> "The platform "linux" is incompatible with this module."

I thought this first, but it doesn't seem so. blueprintjs of course runs on linux and so I eliminated that route for the time being.
What I did then to first switch to V3.43 - the current release - et voila - that one installed without problems. I still don't known, however, what is causing the problem above.

On which platform are you guys building blueprintjs? Apple? Windows? Linux? If it is not Linux, then you might have a point.

P.S.: fsevents seems to be needed for Apple platforms only - see e.g. https://stackoverflow.com/questions/57082249/how-to-fix-error-fsevents2-0-7-the-platform-linux-is-incompatible-with-thi
